source/grid/grid_tools.cc

```diff
@@ -3742,9 +3742,20 @@ namespace GridTools
     Assert (pairs1.size() == pairs2.size(),
             ExcMessage ("Unmatched faces on periodic boundaries"));
 
+#ifdef DEBUG
+    const unsigned int size_old = matched_pairs.size();
+#endif
+
     // and call match_periodic_face_pairs that does the actual matching:
     match_periodic_face_pairs(pairs1, pairs2, direction, matched_pairs, offset,
                               matrix);
+
+#ifdef DEBUG
+    Assert(matched_pairs.size() > size_old,
+           ExcMessage("No new periodic face pairs have been found."
+                      "Are you sure that you've selected the correct boundary"
```

Comment:
You need a whitespace at the end of each line.

Actually, it should be sufficient to check `pairs1.size()>0`, isn't it? This way you don't need all these preprocessor variables and `size_old`.